### PR TITLE
KTOR-5687 Fix serializing with typeInfo Any

### PR DESCRIPTION
**Pull request by e5l**
[KTOR-5687](https://youtrack.jetbrains.com/issue/KTOR-5687)

### CODE REVIEW

**Comment by rsinukov** [comment]
Can you add another test for sealed class? In the ticket there is a mention that it's broken

**Review by Stexxe — APPROVED** [review]

**Comment by e5l** [comment]
Thanks for the notice, sure

**Comment by e5l** [comment]
> Can you add another test for sealed class? In the ticket there is a mention that it's broken

I didn't find mention of sealed in the issue, but will add test anyway

File: ktor-client/ktor-client-plugins/ktor-client-content-negotiation/ktor-client-content-negotiation-tests/jvm/src/io/ktor/client/plugins/contentnegotiation/tests/JsonContentNegotiationTest.kt [hunk 1]
```diff
@@ -231,4 +231,58 @@ abstract class JsonContentNegotiationTest(private val converter: ContentConverte
...
+            }
+        }.get("/").let { response ->
+            assertEquals(HttpStatusCode.OK, response.status)
+            assertEquals("""{"value":"abc"}""", response.bodyAsText())
+        }
+    }
 }
+
+@Serializable
+internal sealed class Data
```
Comment (rsinukov) [inline ktor-client/ktor-client-plugins/ktor-client-content-negotiation/ktor-client-content-negotiation-tests/jvm/src/io/ktor/client/plugins/contentnegotiation/tests/JsonContentNegotiationTest.kt:285]: can you also add a test with serialisation of a sub type?

**Review by rsinukov — APPROVED** [review]

File: ktor-client/ktor-client-plugins/ktor-client-content-negotiation/ktor-client-content-negotiation-tests/jvm/src/io/ktor/client/plugins/contentnegotiation/tests/JsonContentNegotiationTest.kt (hunk 1 above)
Comment (e5l) [reply ktor-client/ktor-client-plugins/ktor-client-content-negotiation/ktor-client-content-negotiation-tests/jvm/src/io/ktor/client/plugins/contentnegotiation/tests/JsonContentNegotiationTest.kt]: sure

**Pull request merged**